storage/namespace_test.go

```diff
@@ -52,36 +53,42 @@ var (
 	testShardIDs = sharding.NewShards([]uint32{0, 1}, shard.Available)
 )
 
-func newTestNamespace(t *testing.T) *dbNamespace {
+type closer func()
+
+func newTestNamespace(t *testing.T) (*dbNamespace, closer) {
 	return newTestNamespaceWithIDOpts(t, defaultTestNs1ID, defaultTestNs1Opts)
 }
 
 func newTestNamespaceWithIDOpts(
 	t *testing.T,
 	nsID ident.ID,
 	opts namespace.Options,
-) *dbNamespace {
+) (*dbNamespace, closer) {
 	metadata, err := namespace.NewMetadata(nsID, opts)
 	require.NoError(t, err)
 	hashFn := func(identifier ident.ID) uint32 { return testShardIDs[0].ID() }
 	shardSet, err := sharding.NewShardSet(testShardIDs, hashFn)
 	require.NoError(t, err)
-	dopts := testDatabaseOptions()
+	dopts := testDatabaseOptions().SetRuntimeOptionsManager(runtime.NewOptionsManager())
 	ns, err := newDatabaseNamespace(metadata, shardSet, nil, nil, nil, nil, dopts)
 	require.NoError(t, err)
-	return ns.(*dbNamespace)
+
+	closer := func() { dopts.RuntimeOptionsManager().Close() }
```

Comment:
nit: can probably just be `closer := dopts.RuntimeOptionsManager.Close`

Reply:
sure thing

Reply:
done